components/bpmn/org.wso2.carbon.bpmn.analytics.publisher/src/main/java/org/wso2/carbon/bpmn/analytics/publisher/BPMNDataPublisher.java

```diff
@@ -446,50 +399,50 @@ public void publishKPIvariableData(ProcessInstance processInstance) throws BPMND
-                {"isAnalyzeData":false,"name":"status","isDrillDownData":false,"type":"string"},
-                {"isAnalyzeData":false,"name":"pizzaTopping","isDrillDownData":false,"type":"string"},
-                {"isAnalyzeData":false,"name":"amount","isDrillDownData":false,"type":"int"},
-                {"isAnalyzeData":"false","name":"processInstanceId","isDrillDownData":"false","type":"string"}]
-                 */
-                JsonArray configedProcVarsJson = kpiConfig
-                        .getAsJsonArray(AnalyticsPublisherConstants.PROCESS_VARIABLES_JSON_ENTRY_NAME);
-
-                int variableCount = configedProcVarsJson.size();
-                configedProcessVariables = new String[variableCount][2];
-
-                for (int i = 0; i < variableCount; i++) {
-                    configedProcessVariables[i][0] = ((JsonObject) configedProcVarsJson.get(i)).get("name")
-                            .getAsString();
-                    configedProcessVariables[i][1] = ((JsonObject) configedProcVarsJson.get(i)).get("type")
-                            .getAsString();
-                }
-
-                processVariablesMap.put(processDefinitionId, configedProcessVariables);
-                eventStreamId = kpiConfig.get("eventStreamId").getAsString();
-                kpiStreamIdMap.put(processDefinitionId, eventStreamId);
+            JsonObject kpiConfig = getKPIConfiguration(processDefinitionId);
+            //do not publish the KPI event if DAS configurations are not done by the PC
+            if (kpiConfig == null) {
+                return;
+            }
+            if (log.isDebugEnabled()) {
+                log.debug(
+                        "Publishing Process Variables (KPI) for the process instance " + processInstanceId + " of the "
+                                + "process : " + processDefinitionId);
+            }
 
-            } else { //if the process variables are already taken, get them from the Map
-                configedProcessVariables = processVariablesMap.get(processDefinitionId);
-                eventStreamId = kpiStreamIdMap.get(processDefinitionId);
+            /* Keeps configured process variable data as a JSON. Example value:
+            [{"name":"processInstanceId","type":"string","isAnalyzeData":"false","isDrillDownData":"false"},
+            {"name":"valuesAvailability","type":"string","isAnalyzeData":"false","isDrillDownData":"false"},
+            {"name":"custid","type":"string","isAnalyzeData":false,"isDrillDownData":false},
+            {"name":"amount","type":"long","isAnalyzeData":false,"isDrillDownData":false},
+            {"name":"confirm","type":"bool","isAnalyzeData":false,"isDrillDownData":false}]
+            */
+            JsonArray fieldsConfigedForStreamPayload = kpiConfig
+                    .getAsJsonArray(AnalyticsPublisherConstants.PROCESS_VARIABLES_JSON_ENTRY_NAME);
+
```

Comment:
As we not storing variable configs in memory anymore, we can directly use jsonarray without converting it to a 2D array.